src/ec/packages/ec-component-media-container/media-container.html.twig

```diff
@@ -0,0 +1,75 @@
+{% spaceless %}
+{#
+  Parameters:
+  - "alt" (string) (default: ''),
+  - "image" (string) (default: ''),
+  - "extra_classes" (string) (default: ''),
+  - "sources" (array) (default: []): format: [
+      {
+        "src" (string) (default: ''),
+        "type" (string) (default: ''),
+      },
+      ...
+    ],
+  - "tracks" (array) (default: []): format: [
+      {
+        "src" (string) (default: ''),
+        "kind" (string) (default: ''),
+        "srcLang" (string) (default: ''),
+        "label" (string) (default: ''),
+      },
+      ...
+    ],
+  - "description" (string) (default: ''),
+  - "extra_attributes" (array) (default: []): format: [
+      {
+        "name" (string) (default: ''),
+        "value" (string) (default: '')
+      },
+      ...
+    ]
+#}
+
+{# Internal properties #}
+
+{% set _css_class = 'ecl-media-container' %}
+{% set _image = image|default('') %}
+{% set _description = description|default('') %}
+{% set _alt = image|default('') %}
+{% set _extra_attributes = '' %}
+{% set _sources = sources|default({}) %}
+{% set _tracks = tracks|default({}) %}
+
+{# Internal logic - Process properties #}
+
+{% if extra_classes is defined and extra_classes is not empty %}
+  {% set _css_class = _css_class ~ ' ' ~ extra_classes %}
+{% endif %}
+
+{% if extra_attributes is defined and extra_attributes is not empty and extra_attributes is iterable %}
+  {% for attr in extra_attributes %}
+    {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name ~ '="' ~ attr.value ~ '"' %}
+  {% endfor %}
+{% endif %} 
+
+{# Print the result #}
+
+<figure class="{{ _css_class }}"{{ _extra_attributes|raw }}>
+{%- if _sources is not empty -%}
```

Comment:
Might worth transferring the checks of `iterable` too (https://github.com/ec-europa/europa-component-library/blob/v2-dev/src/systems/ec/implementations/react/packages/media-container/src/MediaContainer.jsx#L18)